src/simplewallet/simplewallet.cpp

```diff
@@ -1588,7 +1588,17 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
       }
     }
     if (m_restoring)
-      m_wallet->set_refresh_from_block_height(m_restore_height);
+    {
+      uint64_t estimate_height = m_wallet->estimate_blockchain_height();
+      if (m_restore_height >= estimate_height)
+      {
+        success_msg_writer() << tr("Restore height ") << m_restore_height << (" is not yet reached. The current estimated height is ") << estimate_height;
+        std::string confirm = input_line(tr("Still apply restore height?  (Y/Yes/N/No): "));
+        if (std::cin.eof() || command_line::is_no(confirm))
+          m_restore_height = 0;
+        m_wallet->set_refresh_from_block_height(m_restore_height);
```

Comment:
Shouldn't this line go outside this `if` scope?

Reply:
Correct!